lib/GenomeFileUtil/core/GenomeInterface.py

```diff
@@ -128,54 +193,75 @@ def get_one_genome(self, params):
-        # XXX there is no `workspace_datatype` param in the spec
-        ws_datatype = params.get('workspace_datatype', "KBaseGenomes.Genome")
-        # XXX there is no `meta` param in the spec
-        meta = params.get('meta', {})
-        if "AnnotatedMetagenomeAssembly" in ws_datatype:
-            if params.get('upgrade') or 'feature_counts' not in data:
-                data = self._update_metagenome(data)
-        else:
-            if params.get('upgrade') or 'feature_counts' not in data:
-                data = self._update_genome(data)
-
-        # check all handles point to shock nodes owned by calling user
-        self._own_handle(data, 'genbank_handle_ref')
-        self._own_handle(data, 'gff_handle_ref')
-        if "AnnotatedMetagenomeAssembly" not in ws_datatype:
-            self._check_dna_sequence_in_features(data)
-            data['warnings'] = self.validate_genome(data)
-
-        # sort data
-        data = GenomeUtils.sort_dict(data)
-        # dump genome to scratch for upload
-        data_path = os.path.join(self.scratch, name + ".json")
-        json.dump(data, open(data_path, 'w'))
-        if 'hidden' in params and str(params['hidden']).lower() in ('yes', 'true', 't', '1'):
-            hidden = 1
-        else:
-            hidden = 0
-
-        if isinstance(workspace, int) or workspace.isdigit():
-            workspace_id = workspace
-        else:
-            workspace_id = self.dfu.ws_name_to_id(workspace)
-
-        save_params = {'id': workspace_id,
-                       'objects': [{'type': ws_datatype,
-                                    'data_json_file': data_path,
-                                    'name': name,
-                                    'meta': meta,
-                                    'hidden': hidden}]}
-        dfu_oi = self.ws_large_data.save_objects(save_params)[0]
-        returnVal = {'info': dfu_oi, 'warnings': data.get('warnings', [])}
-        return returnVal
+    def _save_genome_mass(self, params):
+
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+
+        ws_datatypes = []
+        data_paths = []
+        names = []
+        meta_data = []
+        hidden_data = []
+        warnings = []
+
+        for input_params in inputs:
+
+            # retrive required params
+            name = input_params['name']
+            data = input_params['data']
+
+            # XXX there is no `workspace_datatype` param in the spec
+            ws_datatype = input_params.get('workspace_datatype', "KBaseGenomes.Genome")
+            # XXX there is no `meta` param in the spec
+            meta = input_params.get('meta', {})
+
+            ws_datatypes.append(ws_datatype)
+            names.append(name)
+            meta_data.append(meta)
+
+            if "AnnotatedMetagenomeAssembly" in ws_datatype:
+                if input_params.get('upgrade') or 'feature_counts' not in data:
+                    data = self._update_metagenome(data)
```

Comment:
Unfortunately, given you changed this code, this line will need to be covered.

Reply:
Ah, in order to test, we need a JSON format metagenomes file. However, such a file does not exist in the https://github.com/kbaseapps/GenomeFileUtil/tree/master/test/data/metagenomes.

Reply:
So add one?

Reply:
👍

Reply:
For the test, there's no need to add tiny files to the repo, just create them in the test as needed

Reply:
Also, it looks like the test only checks a small part of the workpsace info returned from the workspace. It doesn't test the object contents at all

Reply:
~> For the test, there's no need to add tiny files to the repo, just create them in the test as needed~

Reply:
> ~> For the test, there's no need to add tiny files to the repo, just create them in the test as needed~ Done

Reply:
added more info and warnings check

Reply:
Why not just write the files directly to the path in the shared space rather than creating a temp file and then copying it to the shared space?

Reply:
Also, this still needs to be deleted as we discussed

Reply:
> Also, this still needs to be deleted as we discussed

Based on our last discussion, we kept these three lines since I had already added tests to cover the missing line.

Reply:
That's not what I understood - I understood that you should keep any tests you added but remove the unusable code. If that means the tests need to be delete, fine - way better to delete unused code & tests than to keep it around just because it's tested

Reply:
deleted unused code & tests

Reply:
:+1: